### PR TITLE
Fix sorting of Identifiers

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Comment by fviernau** [comment]
@sschuberth why is this a "breaking change" ?

**Comment by sschuberth** [comment]
> @sschuberth why is this a "breaking change" ?

I just wanted to be conservative in case someone has workflow steps that rely on the sorting by `id`, e.g. in the `curations.yml`. We actually have pre-submit checks in CI to enforce alphabetical sorting in that file. And this change would break it.

But I agree it's debatable. Should I remove the "breaking change" label?

**Comment by fviernau** [comment]
> But I agree it's debatable. Should I remove the "breaking change" label?

I'm fine with the label. I just wanted to double check whether I'm not missing anything breaking here.

File: analyzer/src/funTest/kotlin/managers/PubFunTest.kt [hunk 1]
```diff
@@ -108,7 +108,7 @@ class PubFunTest : WordSpec() {
                 result.toYaml() shouldBe expectedResult
             }
 
-            "Error is shown when no lockfile is present" {
+            "shown an error if no lockfile is present" {
```
Comment (oheger-bosch) [inline analyzer/src/funTest/kotlin/managers/PubFunTest.kt:111]: Nit: "shown" -> "show"

**Review by oheger-bosch — APPROVED** [review]

**Review by MarcelBochtler — DISMISSED** [review]

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**